server/queue/queue-jms/src/test/java/org/apache/james/queue/jms/JMSSerializationUtilsTest.java

```diff
@@ -0,0 +1,119 @@
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.queue.jms;
+
+import static org.apache.james.queue.jms.JMSSerializationUtils.deserialize;
+import static org.apache.james.queue.jms.JMSSerializationUtils.hasJMSNativeSupport;
+import static org.apache.james.queue.jms.JMSSerializationUtils.roundtrip;
+import static org.apache.james.queue.jms.JMSSerializationUtils.trySerialize;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.SerializationException;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.Test;
+
+class JMSSerializationUtilsTest {
+    @Test
+    @SuppressWarnings("ConstantConditions")
+    void hasJMSNativeSupportSpec() {
+        SoftAssertions softly = new SoftAssertions();
+
+        softly.assertThat(hasJMSNativeSupport(null)).as("null support").isTrue();
+        softly.assertThat(hasJMSNativeSupport("")).as("String support").isTrue();
+        softly.assertThat(hasJMSNativeSupport((byte) 0)).as("Byte support").isTrue();
+        softly.assertThat(hasJMSNativeSupport((long) 0)).as("Long support").isTrue();
+        softly.assertThat(hasJMSNativeSupport((double) 0)).as("Double support").isTrue();
+        softly.assertThat(hasJMSNativeSupport(0)).as("Integer support").isTrue();
+        softly.assertThat(hasJMSNativeSupport((short) 0)).as("Short support").isTrue();
+        softly.assertThat(hasJMSNativeSupport((float) 0)).as("Float support").isTrue();
+        softly.assertThat(hasJMSNativeSupport(true)).as("Boolean support").isTrue();
+
+        softly.assertAll();
+    }
+
+    @Test
+    void trySerializeShouldReturnItselfWhenJMSSupported() {
+        Integer expected = 41;
+
+        Object actual = trySerialize(expected);
+
+        assertThat(actual).isSameAs(expected);
+    }
+
+    @Test
+    void trySerializeShouldReturnString() {
+        SerializableStringHolder value = new SerializableStringHolder("value");
+
+        Object actual = trySerialize(value);
+
+        assertThat(actual)
+                .isInstanceOf(String.class);
```

Comment:
Can we get an assert on the string content?

Reply:
Wouldn't that just test for test?

Reply:
It would allow while reading the test class to get a glance at what goes on the wire in JMS/ActiveMQ. While not necessary, I consider it allows seeing the length of data, and the structure.